dove/src/stdout.rs

```diff
@@ -0,0 +1,213 @@
+use termcolor::{BufferWriter, Color, ColorChoice, ColorSpec, WriteColor};
+use once_cell::sync::OnceCell;
+use std::fmt::Write as FmtWriter;
+use std::io::{Write, Stdout};
+use std::sync::Mutex;
+use anyhow::Error;
+
+/// Stdout buffer for prints
+static STDOUT_STREAM: OnceCell<Mutex<Box<dyn TBufWrite + Send>>> = OnceCell::new();
+/// Marker to highlight; WHEN is 'always', 'never', or 'auto'
+static STDOUT_COLOR: OnceCell<Mutex<Box<ColorChoice>>> = OnceCell::new();
+
+/// Print output.
+pub fn print(text: &str) -> Result<(), Error> {
+    STDOUT_STREAM
+        .get()
+        .unwrap()
+        .lock()
+        .unwrap()
+        .as_mut()
+        .print(text)
+}
+
+/// Print output with a newline.
+pub fn println(text: &str) -> Result<(), Error> {
+    print(text)?;
+    print("\n")
+}
+
+/// print output in green color
+pub fn print_good(text: &str) -> Result<(), Error> {
+    print_with_color(text, ColorSpec::new().set_fg(Some(Color::Green)))
+}
+
+/// print output in red color
+pub fn print_error(text: &str) -> Result<(), Error> {
+    print_with_color(text, ColorSpec::new().set_fg(Some(Color::Red)))
+}
+
+/// print output in yellow color
+pub fn print_warning(text: &str) -> Result<(), Error> {
+    print_with_color(text, ColorSpec::new().set_fg(Some(Color::Yellow)))
+}
+
+/// print output with color settings
+fn print_with_color(text: &str, spec: &mut ColorSpec) -> Result<(), Error> {
+    let mut buffer = BufferWriter::stdout(get_stdout_color()).buffer();
+    buffer.set_color(spec)?;
+    write!(&mut buffer, "{}", text)?;
+    buffer.reset()?;
+    print(&String::from_utf8_lossy(buffer.as_slice()))
+}
+
+/// Prints output.
+#[macro_export]
+macro_rules! stdout {
+    ($($args:tt)*) => {
+        $crate::stdout::print(&format!($($args)*)).unwrap();
+    };
+}
+
+/// Prints output with a newline.
+#[macro_export]
+macro_rules! stdoutln {
+    () => ($crate::stdout::print!("\n").unwrap());
+    ($($args:tt)*) => {
+        $crate::stdout::println(&format!($($args)*)).unwrap();
+    }
+}
+
+/// Set the stream / buffer to write
+pub fn set_buffer<Out>(stdout: Out) -> Result<(), Error>
+where
+    Out: TBufWrite + Send,
+    Out: 'static,
+{
+    let flag = get_argv_flag_color(&stdout);
+
+    match STDOUT_COLOR.get() {
+        Some(mt) => {
+            let mut buff = mt.lock().unwrap();
+            *buff = Box::new(flag);
+            Ok(())
+        }
+        None => STDOUT_COLOR
+            .set(Mutex::new(Box::new(flag)))
+            .map_or_else(|_| Err(anyhow!("Failed to set STDOUT_COLOR")), |_| Ok(())),
```

Comment:
nit:
```rust
.map_err(|_| anyhow!("Failed to set STDOUT_COLOR"))?;
Ok(())
```
is a bit more readable, what do you think?

Reply:
Unfortunately this doesn't work

Reply:
What do you mean?

Reply:
Done. Thanks